crates/uv-configuration/src/build_options.rs

```diff
@@ -108,7 +108,22 @@ pub enum NoBinary {
 
 impl NoBinary {
     /// Determine the binary installation strategy to use for the given arguments.
-    pub fn from_args(no_binary: Vec<PackageNameSpecifier>) -> Self {
+    pub fn from_args(no_binary: Option<bool>, no_binary_package: Vec<PackageName>) -> Self {
+        match no_binary {
+            Some(true) => Self::All,
+            Some(false) => Self::None,
+            None => {
+                if no_binary_package.is_empty() {
+                    Self::None
+                } else {
+                    Self::Packages(no_binary_package)
+                }
+            }
+        }
+    }
+
+    /// Determine the binary installation strategy to use for the given arguments from the pip CLI.
+    pub fn from_pip_args(no_binary: Vec<PackageNameSpecifier>) -> Self {
```

Comment:
`PackageNameSpecifier` could maybe be renamed to `PipPackageNameSpecifier` since it's only for pip right?

Reply:
I left this as-is for now, I wavered on it...